src/emulator/modules/SceSysmem/SceSysmem.cpp

```diff
@@ -81,8 +95,18 @@ EXPORT(int, sceKernelFreeMemBlock, SceUID uid) {
     return SCE_KERNEL_OK;
 }
 
-EXPORT(int, sceKernelFreeMemBlockForVM) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceKernelFreeMemBlockForVM, SceUID uid) {
+    assert(uid >= 0);
+    
+    KernelState *const state = &host.kernel;
+    const Blocks::const_iterator block = state->blocks.find(uid);
```

Comment:
Since you are pushing vm_blocks explicitly, is it better to search in vm_blocks.
Else just delete the vm_blocks map completely. It has no usage, and just increase memory usage

Reply:
The new map is needed for future functions: open, close and sync domain. I'll make the find change.